packages/components/src/components/base/models/QueryInfo.ts

```diff
@@ -0,0 +1,335 @@
+    // isMetadataOverrideable: false,
+    // isTemporary: false,
+    // isUserDefined: false,
+    lastAction: undefined,
+    // lastUpdate: undefined,
+    name: undefined,
+    pkCols: List<string>(),
+    schemaName: undefined,
+    status: QueryInfoStatus.unknown,
+    // targetContainers: List<any>(),
+    title: undefined, // DEPRECATED: Use queryLabel
+    titleColumn: undefined,
+    // viewDataUrl: undefined,
+    views: Map<string, ViewInfo>(),
+    importUrlDisabled: undefined,
+    importUrl: undefined,
+    insertUrlDisabled: undefined,
+    insertUrl: undefined,
+
+    // our stuff
+    appEditableTable: false,
+    isLoading: false,
+    isMedia: false, // opt in
+    queryLabel: undefined,
+    schemaLabel: undefined,
+    schemaQuery: undefined,
+    showInsertNewButton: true, // opt out
+    singular: undefined, // defaults to value of queryLabel
+    plural: undefined  // defaults to value of queryLabel
+}) {
+    private appEditableTable: boolean; // use isAppEditable()
+    // canEdit: boolean;
+    // canEditSharedViews: boolean;
+    columns: OrderedMap<string, QueryColumn>;
+    description: string;
+    // editDefinitionUrl: string;
+    iconURL: string;
+    importTemplates: List<any>;
+    // indices: Map<string, any>;
+    // isInherited: boolean;
+    isLoading: boolean;
+    isMedia: boolean;
+    // isMetadataOverrideable: boolean;
+    // isTemporary: boolean;
+    // isUserDefined: boolean;
+    lastAction: LastActionStatus;
+    // lastUpdate: Date;
+    name: string;
+    pkCols: List<string>;
+    plural: string;
+    queryLabel: string;
+    schemaName: string;
+    schemaQuery: SchemaQuery;
+    singular: string;
+    status: QueryInfoStatus;
+    // targetContainers: List<any>;
+    title: string;
+    titleColumn: string;
+    // viewDataUrl: string;
+    views: Map<string, ViewInfo>;
+    schemaLabel: string;
+    showInsertNewButton: boolean;
+    importUrlDisabled: boolean;
+    importUrl: string;
+    insertUrlDisabled: boolean;
+    insertUrl: boolean;
+
+    static create(rawQueryInfo: any): QueryInfo {
+        let schemaQuery: SchemaQuery;
+
+        if (rawQueryInfo.schemaName && rawQueryInfo.name) {
+            schemaQuery = SchemaQuery.create(rawQueryInfo.schemaName, rawQueryInfo.name);
+        }
+
+        return new QueryInfo(Object.assign({}, rawQueryInfo, {
+            schemaQuery
+        }));
+    }
+
+    /**
+     * Use this method for creating a basic QueryInfo object with a proper schemaQuery object
+     * and columns map from a JSON object.
+     *
+     * @param queryInfoJson
+     */
+    static fromJSON(queryInfoJson: any) : QueryInfo {
+        let schemaQuery: SchemaQuery;
+
+        if (queryInfoJson.schemaName && queryInfoJson.name) {
+            schemaQuery = SchemaQuery.create(queryInfoJson.schemaName, queryInfoJson.name);
+        }
+        let columns = OrderedMap<string, QueryColumn>();
+        Object.keys(queryInfoJson.columns).forEach((columnKey) => {
```

Comment:
Object.values()?

Reply:
I just copy/pasted this model and don't plan to make any modifications other than the one method I added.